core/src/test/java/org/frankframework/pipes/EscapePipeTest.java

```diff
@@ -51,6 +51,44 @@ public void testPartialEscapedMultiLineXmlInput() throws Exception { //double es
 		TestAssertions.assertEqualsIgnoreCRLF(expected, result.getResult().asString());
 	}
 
+	@Test
+	public void testPartialEscapedMultiLineXmlInputMultiOccurences() throws Exception {
+		pipe.setSubstringStart("<message>");
+		pipe.setSubstringEnd("</message>");
+		configureAndStartPipe();
+
+		PipeRunResult result = doPipe(getResource("multi-line-partial-escaped-multi-occurences.xml"));
+		String expected = getResource("multi-line-partial-escaped-multi-occurences.result").asString();
+		TestAssertions.assertEqualsIgnoreCRLF(expected, result.getResult().asString());
+	}
+
+	@Test
+	public void testPartialEscapedMultiLineXmlInputMultiOccurencesMissingClosingTag() throws Exception {
+		pipe.setSubstringStart("<message>");
+		pipe.setSubstringEnd("</message>");
+		configureAndStartPipe();
+
+		PipeRunResult result = doPipe(getResource("no-closing-tag-configured.xml"));
+		String expected = getResource("no-closing-tag-configured.result").asString();
```

Comment:
The second <message> opening is missing the </message> closing tag here right?

Reply:
The filename is still confusing because closing tag is configured in the pipe, but not supplied in the input.
The filename implies the closing tag is not configured. Configuration applies to the pipe.